package/MDAnalysis/coordinates/RDKit.py

```diff
@@ -84,21 +126,708 @@ def _format_hint(thing):
+        # the atom
+        if (len(nue) == 1) and (nue[0] < 0):
+            atom.SetFormalCharge(-nue[0])
+            mol.UpdatePropertyCache(strict=False)
+        # go to next atom if above case or atom has no unpaired electron
+        if (len(nue) == 1) and (nue[0] <= 0):
+            continue
+        else:
+            neighbors = sorted(atom.GetNeighbors(), reverse=True,
+                               key=lambda a: _get_nb_unpaired_electrons(a)[0])
+            # check if one of the neighbors has a common NUE
+            for i, na in enumerate(neighbors, start=1):
+                # get NUE for the neighbor
+                na_nue = _get_nb_unpaired_electrons(na)
+                # smallest common NUE
+                common_nue = min(
+                    min([i for i in nue if i >= 0], default=0),
+                    min([i for i in na_nue if i >= 0], default=0)
+                )
+                # a common NUE of 0 means we don't need to do anything
+                if common_nue != 0:
+                    # increase bond order
+                    bond = mol.GetBondBetweenAtoms(
+                        atom.GetIdx(), na.GetIdx())
+                    order = common_nue + 1
+                    bond.SetBondType(RDBONDORDER[order])
+                    mol.UpdatePropertyCache(strict=False)
+                    if i < len(neighbors):
+                        # recalculate nue for atom
+                        nue = _get_nb_unpaired_electrons(atom)
+
+            # if the atom still has unpaired electrons
+            nue = _get_nb_unpaired_electrons(atom)[0]
+            if nue > 0:
+                # transform it to a negative charge
+                atom.SetFormalCharge(-nue)
+                atom.SetNumRadicalElectrons(0)
+                mol.UpdatePropertyCache(strict=False)
+
+
+def _get_nb_unpaired_electrons(atom):
+    """Calculate the number of unpaired electrons (NUE) of an atom
+
+    Parameters
+    ----------
+    atom: rdkit.Chem.rdchem.Atom
+        The atom for which the NUE will be computed
+
+    Returns
+    -------
+    nue : list
+        The NUE for each possible valence of the atom
+    """
+    expected_vs = PERIODIC_TABLE.GetValenceList(atom.GetAtomicNum())
+    current_v = atom.GetTotalValence() - atom.GetFormalCharge()
+    return [v - current_v for v in expected_vs]
+
+
+def _standardize_patterns(mol, max_iter=200):
+    """Standardizes functional groups
+
+    Uses :func:`_rebuild_conjugated_bonds` to standardize conjugated systems,
+    and SMARTS reactions for other functional groups.
+    Due to the way reactions work, we first have to split the molecule by
+    fragments. Then, for each fragment, we apply the standardization reactions.
+    Finally, the fragments are recombined.
+
+    Notes
+    -----
+    The following functional groups are transformed:
+
+    +---------------+-------------------------------------------------------------------------+
```

Comment:
Awesome, this is exactly what we needed here. However, I think that usually we have Parameters & Returns before Notes. Also could you add a line that mentions that these operations are done in order (I doubt there's much overlap, but this is really in case someone has something crazy that just gets overridden by all this).